File: src/Application Specific/Classes/Entity Collections/Quotes/QuoteContractsOriginal/QuoteContractOriginalSiteAsset/QuoteContractOriginalSiteAssetSQL.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace QuoteContractOriginalSiteAssets

        Public Class QuoteContractOriginalSiteAssetSQL
            Private _database As Sys.Database

            Public Sub New(ByVal database As Sys.Database)
                _database = database
            End Sub

#Region "Functions"


            Public Sub Delete(ByVal QuoteContractSiteID As Integer)
                _database.ClearParameter()
                _database.AddParameter("@QuoteContractSiteID", QuoteContractSiteID, True)
                _database.ExecuteSP_NO_Return("QuoteContractOriginalSiteAsset_Delete")
            End Sub

            Public Function QuoteContractSiteAsset_GetAll_QuoteContractSiteID(ByVal QuoteContractSiteID As Integer, _
                                                                            ByVal SiteID As Integer) As DataView
                _database.ClearParameter()
                _database.AddParameter("@QuoteContractSiteID", QuoteContractSiteID, True)
                _database.AddParameter("@SiteID", SiteID, True)
                Dim dt As DataTable = _database.ExecuteSP_DataTable("QuoteContractOriginalSiteAsset_GetAll_QuoteContractSiteID")
                dt.TableName = Entity.Sys.Enums.TableNames.tblQuoteContractSiteAsset.ToString
                Return New DataView(dt)
            End Function

            Public Function [Insert](ByVal oQuoteContractSiteAsset As QuoteContractOriginalSiteAsset) As QuoteContractOriginalSiteAsset
                _database.ClearParameter()
                AddQuoteContractSiteAssetParametersToCommand(oQuoteContractSiteAsset)

                oQuoteContractSiteAsset.SetQuoteContractSiteAssetID = Entity.Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("QuoteContractOriginalSiteAsset_Insert"))
                oQuoteContractSiteAsset.Exists = True
                Return oQuoteContractSiteAsset
            End Function

            Private Sub AddQuoteContractSiteAssetParametersToCommand(ByRef oQuoteContractSiteAsset As QuoteContractOriginalSiteAsset)
                With _database
                    .AddParameter("@QuoteContractSiteID", oQuoteContractSiteAsset.QuoteContractSiteID, True)
                    .AddParameter("@AssetID", oQuoteContractSiteAsset.AssetID, True)
                    .AddParameter("@PricePerVisit", oQuoteContractSiteAsset.PricePerVisit, True)



                End With
            End Sub


#End Region

        End Class

    End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Quotes/QuoteContractsOriginal/QuoteContractOriginalSiteAsset/QuoteContractOriginalSiteAssetSQL.cs
using System.Data;

namespace FSM.Entity
{
    namespace QuoteContractOriginalSiteAssets
    {
        public class QuoteContractOriginalSiteAssetSQL
        {
            private Sys.Database _database;

            public QuoteContractOriginalSiteAssetSQL(Sys.Database database)
            {
                _database = database;
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */

            public void Delete(int QuoteContractSiteID)
            {
                _database.ClearParameter();
                _database.AddParameter("@QuoteContractSiteID", QuoteContractSiteID, true);
                _database.ExecuteSP_NO_Return("QuoteContractOriginalSiteAsset_Delete");
            }

            public DataView QuoteContractSiteAsset_GetAll_QuoteContractSiteID(int QuoteContractSiteID, int SiteID)
            {
                _database.ClearParameter();
                _database.AddParameter("@QuoteContractSiteID", QuoteContractSiteID, true);
                _database.AddParameter("@SiteID", SiteID, true);
                var dt = _database.ExecuteSP_DataTable("QuoteContractOriginalSiteAsset_GetAll_QuoteContractSiteID");
                dt.TableName = Sys.Enums.TableNames.tblQuoteContractSiteAsset.ToString();
                return new DataView(dt);
            }

            public QuoteContractOriginalSiteAsset Insert(QuoteContractOriginalSiteAsset oQuoteContractSiteAsset)
            {
                _database.ClearParameter();
                AddQuoteContractSiteAssetParametersToCommand(ref oQuoteContractSiteAsset);
                oQuoteContractSiteAsset.SetQuoteContractSiteAssetID = Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("QuoteContractOriginalSiteAsset_Insert"));
                oQuoteContractSiteAsset.Exists = true;
                return oQuoteContractSiteAsset;
            }

            private void AddQuoteContractSiteAssetParametersToCommand(ref QuoteContractOriginalSiteAsset oQuoteContractSiteAsset)
            {
                {
                    var withBlock = _database;
                    withBlock.AddParameter("@QuoteContractSiteID", oQuoteContractSiteAsset.QuoteContractSiteID, true);
                    withBlock.AddParameter("@AssetID", oQuoteContractSiteAsset.AssetID, true);
                    withBlock.AddParameter("@PricePerVisit", oQuoteContractSiteAsset.PricePerVisit, true);
                }
            }


            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}